File: src/main/java/nu/peg/slack/pt/di/factory/OauthServiceFactory.java
package nu.peg.slack.pt.di.factory;

import nu.peg.slack.pt.service.internal.DefaultOauthService;
import nu.peg.slack.pt.service.OauthService;

import org.glassfish.hk2.api.Factory;

import static nu.peg.slack.pt.App.config;

public class OauthServiceFactory implements Factory<OauthService> {

    @Override
    public OauthService provide() {
        String clientId = config.getProperty("slack.client.id");
        String clientSecret = config.getProperty("slack.client.secret");

        return new DefaultOauthService(clientId, clientSecret);
    }

    @Override
    public void dispose(OauthService oauthService) {
    }
}

File: src/main/kotlin/nu/peg/slack/pt/di/factory/OauthServiceFactory.kt
package nu.peg.slack.pt.di.factory

import nu.peg.slack.pt.service.OauthService
import nu.peg.slack.pt.service.internal.DefaultOauthService

import org.glassfish.hk2.api.Factory

import nu.peg.slack.pt.App.config

class OauthServiceFactory : Factory<OauthService> {

    fun provide(): OauthService {
        val clientId = config.getProperty("slack.client.id")
        val clientSecret = config.getProperty("slack.client.secret")

        return DefaultOauthService(clientId, clientSecret)
    }

    fun dispose(oauthService: OauthService) {}
}
